src/lib/data.js

```diff
@@ -37,20 +38,32 @@ export async function getProfileData(id, accessToken, loggedInUsername) {
   const repos = await fetchReposForProfile(
     profile,
     accessToken,
-    loggedInUsername,
-  ).then(repos =>
-    Promise.all(
+    options.loggedInUsername,
+  ).then(repos => {
+    return Promise.all(
       repos.map(async repo => {
-        const dependencies = await getDependenciesFromGithubRepo(
-          repo,
-          accessToken,
-        );
-        // eslint-disable-next-line require-atomic-updates
-        repo.dependencies = dependencies;
+        if (options.excludedRepos) {
+          if (options.excludedRepos.indexOf(repo.name) === -1) {
+            repo.dependencies = await getDependenciesFromGithubRepo(
+              repo,
+              accessToken,
+            );
+            repo.checked = true;
+          } else {
+            repo.checked = false;
+            repo.dependencies = [];
+          }
+        } else {
+          repo.dependencies = await getDependenciesFromGithubRepo(
+            repo,
+            accessToken,
+          );
+          repo.checked = true;
+        }
```

Comment:
This block can be simplified so we don't have the duplicated code section and nested if/else.